mlflow/projects/__init__.py

```diff
@@ -671,10 +700,41 @@ def _validate_docker_installation():
                                  "at https://docs.docker.com/install/overview/.")
 
 
-def _validate_docker_env(docker_env):
-    if not docker_env.get('image'):
+def _validate_docker_env(project):
+    if not project.name:
+        raise ExecutionException("Project name in MLProject must be specified when using docker "
+                                 "for image tagging.")
+    if not project.docker_env.get('image'):
         raise ExecutionException("Project with docker environment must specify the docker image "
-                                 "to use via an 'image' field under the 'docker_env' field")
+                                 "to use via an 'image' field under the 'docker_env' field.")
+
+
+def _parse_kubernetes_config(backend_config, work_dir):
+    """
+    Creates build context tarfile containing Dockerfile and project code, returning path to tarfile
+    """
+    kube_config = backend_config.copy()
+    if not backend_config:
```

Comment:
Will this line ever run? Seems like if `backend_config` is none, the previous line will throw.

Reply:
you're right, I've changed it.